### PR TITLE
Support a list of authorizations

### PR DESCRIPTION
**Pull request by joejstuart**
This updates the authorizations to support a list of repositories and changeIds. Each repo and changeId is compared against the repository and git sha in the attestation.

### CODE REVIEW

File: policy/release/authorization.rego [hunk 1]
```diff
@@ -20,7 +20,7 @@ import data.lib
 #   failure_msg: Commit does not contain authorization
```
Comment (zregvart) [inline policy/release/authorization.rego:20]: Would this be a bit more descriptive?

```suggestion
#   failure_msg: No authorization data found
```

File: policy/release/authorization.rego [hunk 2]
```diff
@@ -45,11 +46,22 @@ deny[result] {
...
-#   failure_msg: Repo url %s does not match authorized repo url %s
+#   failure_msg: Repo url %s does not match authorized repo urls
 deny[result] {
 	data.authorization
+	count(data.authorization) > 0
 	att := lib.pipelinerun_attestations[_]
 	material := att.predicate.materials[_]
-	data.authorization.repository != material.uri
-	result := lib.result_helper(rego.metadata.chain(), [material.uri, data.authorization.repository])
+	not repo_in_auth(material.uri, data.authorization)
```
Comment (zregvart) [inline policy/release/authorization.rego:55]: (nit) could be done without the helper method

```suggestion
	not material.uri in [u | a = data.authorization[_]; u = a.repoUrl]
```

File: policy/release/authorization.rego [hunk 3]
```diff
@@ -30,13 +30,14 @@ deny[result] {
...
-#   failure_msg: Commit %s does not match authorized commit %s
+#   failure_msg: Commit %s does not match authorized commits
 deny[result] {
 	data.authorization
+	count(data.authorization) > 0
 	att := lib.pipelinerun_attestations[_]
 	material := att.predicate.materials[_]
-	data.authorization.changeId != material.digest.sha1
-	result := lib.result_helper(rego.metadata.chain(), [material.digest.sha1, data.authorization.changeId])
+	not sha_in_auth(material.digest.sha1, data.authorization)
```
Comment (zregvart) [inline policy/release/authorization.rego:39]: (nit) could be done without the helper method

```suggestion
	not material.digest.sha1 in [c | a = data.authorization[_]; c = a.changeid]
```

File: policy/release/authorization.rego [hunk 4]
```diff
@@ -30,13 +30,14 @@ deny[result] {
 #   Enterprise Contract verifies if an authorized commit was used as the source of a build
 # custom:
 #   short_name: disallowed_commit_does_not_match
-#   failure_msg: Commit %s does not match authorized commit %s
+#   failure_msg: Commit %s does not match authorized commits
```
Comment (zregvart) [inline policy/release/authorization.rego:33]: Wondering if we would like to say what commits are authorized here for a bit better UX

File: policy/release/authorization.rego [hunk 5]
```diff
@@ -45,11 +46,22 @@ deny[result] {
 #   Enterprise Contract verifies if an authorized repo url was used to build an image
 # custom:
 #   short_name: disallowed_repo_url_does_not_match
-#   failure_msg: Repo url %s does not match authorized repo url %s
+#   failure_msg: Repo url %s does not match authorized repo urls
```
Comment (zregvart) [inline policy/release/authorization.rego:49]: Perhaps saying what repository URLs are authorized would be better UX?

**Review by zregvart — APPROVED** [review]

File: policy/release/authorization.rego (hunk 3 above)
Comment (joejstuart) [reply policy/release/authorization.rego:39]: For some reason I get `unexpected ident token: expected \n or ; or }` when I add this.

Comment (zregvart) [reply policy/release/authorization.rego:39]: Odd, I [tried](https://play.openpolicyagent.org/p/UlT1jRG3fO) it in the playground, it’s really a nit, no need to block the merge

Comment (joejstuart) [reply policy/release/authorization.rego:39]: Even more odd that it doesn't complain when I add the filter to the other function.

Comment (zregvart) [reply policy/release/authorization.rego:39]: Let’s merge, we can refactor when/if needed

**Pull request merged**